ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,295 @@
+// creating and peristing a new one if needed
+func (ses *secureEnclaveSigner) Public() crypto.PublicKey {
+	k, err := ses.currentConsoleUserKey(context.TODO())
+	if err != nil {
+		ses.slogger.Log(context.TODO(), slog.LevelError,
+			"getting public key",
+			"err", err,
+		)
+		return nil
+	}
+
+	return k
+}
+
+func (ses *secureEnclaveSigner) Type() string {
+	return "secure_enclave"
+}
+
+func (ses *secureEnclaveSigner) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+type keyData struct {
+	Uid    string `json:"uid"`
+	PubKey string `json:"pub_key"`
+}
+
+func (ses *secureEnclaveSigner) currentConsoleUserKey(ctx context.Context) (*ecdsa.PublicKey, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	ses.mux.Lock()
+	defer ses.mux.Unlock()
+
+	cu, err := firstConsoleUser(ctx)
+	if err != nil {
+		traces.SetError(span, fmt.Errorf("getting first console user: %w", err))
+		return nil, fmt.Errorf("getting first console user: %w", err)
+	}
+
+	key, ok := ses.uidPubKeyMap[cu.Uid]
+	if ok {
+		span.AddEvent("found_existing_key_for_console_user")
+		return key, nil
+	}
+
+	key, err = ses.createKey(ctx, cu)
+	if err != nil {
+		traces.SetError(span, fmt.Errorf("creating key: %w", err))
+		return nil, fmt.Errorf("creating key: %w", err)
+	}
+
+	span.AddEvent("created_new_key_for_console_user")
+
+	ses.uidPubKeyMap[cu.Uid] = key
+	if err := ses.save(); err != nil {
+		delete(ses.uidPubKeyMap, cu.Uid)
+		traces.SetError(span, fmt.Errorf("saving secure enclave signer: %w", err))
+		return nil, fmt.Errorf("saving secure enclave signer: %w", err)
+	}
+
+	span.AddEvent("saved_key_for_console_user")
+	return key, nil
+}
+
+func (ses *secureEnclaveSigner) MarshalJSON() ([]byte, error) {
+	var keyDatas []keyData
+
+	for uid, pubKey := range ses.uidPubKeyMap {
+		pubKeyBytes, err := echelper.PublicEcdsaToB64Der(pubKey)
+		if err != nil {
+			return nil, fmt.Errorf("converting public key to b64 der: %w", err)
+		}
+
+		keyDatas = append(keyDatas, keyData{
+			Uid:    uid,
+			PubKey: string(pubKeyBytes),
+		})
+
+	}
+
+	return json.Marshal(keyDatas)
+}
+
+func (ses *secureEnclaveSigner) UnmarshalJSON(data []byte) error {
+	if ses.uidPubKeyMap == nil {
+		ses.uidPubKeyMap = make(map[string]*ecdsa.PublicKey)
+	}
+
+	var keyDatas []keyData
+	if err := json.Unmarshal(data, &keyDatas); err != nil {
+		return fmt.Errorf("unmarshalling key data: %w", err)
+	}
+
+	for _, kd := range keyDatas {
+		pubKey, err := echelper.PublicB64DerToEcdsaKey([]byte(kd.PubKey))
+		if err != nil {
+			return fmt.Errorf("converting public key to ecdsa: %w", err)
+		}
+
+		ses.uidPubKeyMap[kd.Uid] = pubKey
+	}
+
+	return nil
+}
+
+func (ses *secureEnclaveSigner) createKey(ctx context.Context, u *user.User) (*ecdsa.PublicKey, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	cmd, err := allowedcmd.Launchctl(
+		ctx,
+		"asuser",
+		u.Uid,
+		"sudo",
+		"--preserve-env",
```

Comment:
Hrm. No idea what the risk is here. It's probably fine, but it's a bit of a yellow flag

Reply:
I believe we need this to retain the os variable `LAUNCHER_SKIP_UPDATES`, though we could move the cmd to the top of main and bypass the update logic completely.

Reply:
There are a couple of ways to approach this.

If that's the only one we care about, maybe we could `--preserve-env= LAUNCHER_SKIP_UPDATES`.

I think we could also invoke sudo with `LAUNCHER_SKIP_UPDATES=true` in the command line. from the man page:
> Environment variables to be set for the command may also be passed as options to sudo in the form    VAR=value, for example LD_LIBRARY_PATH=/usr/local/pkg/lib.